drivers/include/mfrc522.h

```diff
@@ -0,0 +1,844 @@
+ *
+ * MIFARE Restore copies the value of the addressed block into a volatile memory.
+ * For MIFARE Classic only. The sector containing the block must be
+ * authenticated before calling this function. Only for blocks in "value block"
+ * mode, i.e. with access bits [C1 C2 C3] = [110] or [001]. Use
+ * mfrc522_mifare_transfer() to store the result in a block.
+ *
+ * @param[in] dev         Device descriptor of the MFRC522
+ * @param[in] block_addr  Block (0-0xff) number
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_restore(mfrc522_t *dev, uint8_t block_addr);
+
+/**
+ * @brief Write the value from the Transfer Buffer into destination block
+ *
+ * MIFARE Transfer writes the value stored in the volatile memory into one
+ * MIFARE Classic block. For MIFARE Classic only. The sector containing the
+ * block must be authenticated before calling this function. Only for blocks in
+ * "value block" mode, ie with access bits [C1 C2 C3] = [110] or [001].
+ *
+ * @param[in] dev         Device descriptor of the MFRC522
+ * @param[in] block_addr  Block (0-0xff) number
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_transfer(mfrc522_t *dev, uint8_t block_addr);
+
+/**
+ * @brief Helper routine to read the current value from a Value Block
+ *
+ * @note  Only for MIFARE Classic and only for blocks in "value block" mode,
+ *        that is with access bits [C1 C2 C3] = [110] or [001]. The sector
+ *        containing the block must be authenticated before calling this function.
+ *
+ * @param[in]  dev         Device descriptor of the MFRC522
+ * @param[in]  block_addr  Block (0x00-0xff) number
+ * @param[out] value       Current value of the Value Block
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+  */
+mfrc522_status_code_t mfrc522_mifare_get_value(mfrc522_t *dev,
+                                               uint8_t block_addr, int32_t *value);
+
+/**
+ * @brief Helper routine to write a specific value into a Value Block
+ *
+ * @note  Only for MIFARE Classic and only for blocks in "value block" mode,
+ *        that is with access bits [C1 C2 C3] = [110] or [001]. The sector
+ *        containing the block must be authenticated before calling this function.
+ *
+ * @param[in] dev         Device descriptor of the MFRC522
+ * @param[in] block_addr  Block (0x00-0xff) number
+ * @param[in] value       New value of the Value Block
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_set_value(mfrc522_t *dev,
+                                               uint8_t block_addr, int32_t value);
+
+/**
+ * @brief Authenticate with a NTAG216
+ *
+ * @note  Only for NTAG216
+ *
+ * @param[in]  dev       Device descriptor of the MFRC522
+ * @param[in]  password  password
```

Comment:
Looking at the code `password` should be at least 4 bytes long (and exactly 4 bytes are used), I think we should document this.